Sources/Swift/Protocol/Codable/SentryUserCodable.swift

```diff
@@ -0,0 +1,34 @@
+@_implementationOnly import _SentryPrivate
+import Foundation
+
+extension User: Decodable {
+    
+    enum CodingKeys: String, CodingKey {
+        case userId = "id"
+        case email
+        case username
+        case ipAddress = "ip_address"
+        case segment
+        case name
+        case geo
+        case data
+    }
+    
+    @available(*, deprecated, message: "Segment is deprecated, but we still need decode it.")
```

Comment:
`l`: I am not sure if I understand the warning, why do we implement a decoding method which is already deprecated?

Reply:
I'm sorry, I didn't update the message accordingly. This is a bit complicated. Is it clearer now, @philprime? If you have a better suggestion on how to suppress the deprecation error, please let me know.

Reply:
I get it now, for future reference how it looks like without the deprecation availability annotation

![image](https://github.com/user-attachments/assets/67898afd-7be6-4f94-93f2-eafff8147081)

Reply:
Yes exactly.